projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -4297,14 +4297,29 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
     }
 
     /**
-     * Manually marks the `IgxGridComponent` for change detection.
+     * Triggers change detection for the `IgxGridComponent`.
+     * Calling markForCheck also triggers the grid pipes explicitly, resulting in all updates being processed.
+     * May degrade performance if used when not needed, or if misused:
+     * ```typescript
+     * // not needed to use, because the grid updates itself succesfully when a primitive has changed:
+     * grid.data.forEach(rec => {
+     *  rec = newValue;
+     *  grid.markForCheck();
+     * });
+     * // if updating a nested property, Angular will not update the view, need to call markForCheck after the loop:
+     * grid.data.forEach(rec => {
+     *  rec.nestedProp1.nestedProp2 = newValue;
+     * });
+     * grid.markForCheck();
+     * ```
      *
      * @example
      * ```typescript
-     * this.grid1.markForCheck();
+     * grid.markForCheck();
      * ```
      */
     public markForCheck() {
+        this.pipeTrigger++;
```

Comment:
There's quite a few properties and methods where both `pipeTrigger++` and `markForCheck()` are called. Now there's redundant pipe trigger increments.

Reply:
Internally the code base uses `grid.cdr.markForCheck` and not `grid.markForCheck`. There were few places calling directly `markForCheck` that this PR changes to use `cdr.detectChanges()` in order to avoid pipeTrigger incremention.